tfjs-vis/src/render/heatmap.ts

```diff
@@ -252,6 +248,54 @@ export async function heatmap(
   await embed(drawArea, spec, embedOpts);
 }
 
+async function convertToRowMajor(inputValues: number[][]|
+                                 tf.Tensor2D): Promise<number[][]> {
+  let originalShape: number[];
+  let transposed: tf.Tensor2D;
+  if (inputValues instanceof tf.Tensor) {
+    originalShape = inputValues.shape;
+    transposed = inputValues.transpose();
+  } else {
+    originalShape = [inputValues.length, inputValues[0].length];
+    transposed = tf.tidy(() => tf.tensor2d(inputValues).transpose());
+  }
+
+  assert(
+      transposed.rank === 2,
+      'Input to renderHeatmap must be a 2d array or Tensor2d');
+
+  // Download the intermediate tensor values and
+  // dispose the transposed tensor.
+  const transposedValues = await transposed.array();
+  transposed.dispose();
```

Comment:
do you have memory unit tests for this stuff?

Reply:
Yep!